File: src/main/java/br/com/alura/forum/security/configuration/AdminSecurityConfiguration.java
package br.com.alura.forum.security.configuration;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.annotation.Order;
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

import br.com.alura.forum.security.service.UsersService;

@Order(1)
@Configuration
@EnableWebSecurity
public class AdminSecurityConfiguration extends WebSecurityConfigurerAdapter {
	@Autowired
	private UsersService usersService;

	@Override
	protected void configure(HttpSecurity http) throws Exception {
		http.antMatcher("/admin/**")
			.authorizeRequests().anyRequest().hasRole("ADMIN")
		.and()
			.httpBasic();
	}
	
	@Override
	protected void configure(AuthenticationManagerBuilder auth) throws Exception {
		auth.userDetailsService(this.usersService)
		.passwordEncoder(new BCryptPasswordEncoder());
	}
}

File: src/main/kotlin/br/com/alura/forum/security/configuration/AdminSecurityConfiguration.kt
package br.com.alura.forum.security.configuration

import br.com.alura.forum.security.service.UsersService
import org.springframework.context.annotation.Configuration
import org.springframework.core.annotation.Order
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder

@Order(1)
@Configuration
@EnableWebSecurity
class AdminSecurityConfiguration(private val usersService: UsersService) : WebSecurityConfigurerAdapter() {
    override fun configure(http: HttpSecurity) {
        http.antMatcher("/admin/**")
                    .authorizeRequests().anyRequest().hasRole("ADMIN")
                .and()
                    .httpBasic()
    }

    override fun configure(auth: AuthenticationManagerBuilder) {
        auth.userDetailsService(usersService)
                .passwordEncoder(BCryptPasswordEncoder())
    }
}